### ISSUE TITLE
Resize operator output

### ISSUE THREAD

**Issue by cchudant** [sonos/tract#743]
Hello!
I have some trouble with the ONNX Resize operator.

Short list:
- Does not support mode: nearest & nearest_modes like floor, ceil, round_prefer_ceil, round_prefer_floor (Fixed by #723)
- Does not support coordinate_transformation_mode: pytorch_half_pixel
- If one of the output dimension is < than the input dimension, it won't resize on that axis (this is the reason msot of the onnx tests don't pass) -- see https://github.com/sonos/tract/blob/main/onnx/src/ops/resize.rs#L146
- if the optional `roi` input is not present, inference_rules panics
- output_facts panics for some of the onnx tests

I am currently trying to solve all (or at least, most) of these issues in a PR, I think this operator needs a bit of a refactor.
Before that though, Are you ok with me working on that? Also, should this operator be moved to onnx-opl too, for nnef support?

**Comment by kali** [comment]
Yeah, Resize is typically an operator for which we ourselves have very little interest, so we welcome contribution on it.

I am aware of the limitations. I took an axis-per-axis approach that simplifies code a bit but requires two passes. I only looked at the growing case, not the shrinking case. And I only implemented a handful of the various possible combination.

I have not looked at all at the roi stuff... no big surprise.

Migrating it to -opl would be super nice too. I did not bother doing it because it was so uncooked.

If you plan some heavy refactoring, please give me an opportunity to discuss what you intend to do before throwing yourself at it.

**Comment by cchudant** [comment]
Cool!
By refactor I did not mean heavy refactoring, just changing how inference_rules, output_facts are done, moving stuff to opl, and yes, probably redo the evaluation part.

I haven't really thought about how to do Resize without looping on every axes like this yet, but it shouldn't be too hard.

**Comment by mpekala23** [comment]
Hello! I've been running into some issues with Resize when trying to run the [PaddleOCR](https://github.com/PaddlePaddle/PaddleOCR) model in `tract` after [converting it to onnx](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/deploy/paddle2onnx/readme.md). The onnx model works fine in python, but is panicking over an unknown symbol in the resize operator in `tract`. Were there any further developments on this? I'm considering setting up a minimal repo to reproduce and opening an issue, but want to make sure I'm not missing anything from earlier improvements which may fix my problem.

**Comment by kali** [comment]
Hello, thanks for your interest in tract.

No, there has not been any recent development on the Resize front. In order to document the issue, a minimal repo is good, but alternatively, you can use tract command line (see https://github.com/sonos/tract/blob/main/doc/cli-recipe.md#running-a-test-case).

**Comment by kali** [comment]
Resize is implemented (onnx/src/ops/resize.rs). Closing as resolved — please reopen with a repro on a current release if something's still off.
